src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -1370,17 +1376,21 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
                         
             responseRows.add(responseRow);
         }
-
-        if (isFirstGroupedByGiver) {
-            responseRows.addAll(buildMissingResponseRowsBetweenGiverAndPossibleRecipients(
-                                            question, possibleParticipantsWithoutResponses, participantIdentifier,
-                                            bundle.getNameForEmail(participantIdentifier),
-                                            bundle.getTeamNameForEmail(participantIdentifier)));
-        } else {
-            responseRows.addAll(buildMissingResponseRowsBetweenRecipientAndPossibleGivers(
-                                            question, possibleParticipantsWithoutResponses, participantIdentifier,
-                                            bundle.getNameForEmail(participantIdentifier),
-                                            bundle.getTeamNameForEmail(participantIdentifier)));
+        
+        if (isMissingResponsesShown) {
+            if (isFirstGroupedByGiver) {
+                responseRows.addAll(buildMissingResponseRowsBetweenGiverAndPossibleRecipients(
+                                                question, possibleParticipantsWithoutResponses,
+                                                participantIdentifier,
+                                                bundle.getNameForEmail(participantIdentifier),
+                                                bundle.getTeamNameForEmail(participantIdentifier)));
+            } else {
+                responseRows.addAll(buildMissingResponseRowsBetweenRecipientAndPossibleGivers(
+                                                question, possibleParticipantsWithoutResponses,
+                                                participantIdentifier,
+                                                bundle.getNameForEmail(participantIdentifier),
+                                                bundle.getTeamNameForEmail(participantIdentifier)));
```

Comment:
To make it uniform, let's change these two as well even though they look fine as is.